src/env/default.rs

```diff
@@ -257,12 +277,113 @@ impl WriteExt for LogFile {
     }
 }
 
+pub struct AioContext {
+    inner: Option<Arc<LogFd>>,
+    index: usize,
+    aio_vec: Vec<aiocb>,
+    pub(crate) buf_vec: Vec<Vec<u8>>,
+}
+impl AioContext {
+    pub fn new(block_sum: usize) -> Self {
+        let mut aio_vec = vec![];
+        let buf_vec = vec![];
+        unsafe {
+            for _ in 0..block_sum {
+                aio_vec.push(mem::zeroed::<libc::aiocb>());
+            }
+        }
+        Self {
+            inner: None,
+            index: 0,
+            aio_vec,
+            buf_vec,
+        }
+    }
+
+    pub fn set_fd(&mut self, fd: Arc<LogFd>) {
```

Comment:
I understand this is to hide the `LogFd`. But it isn't necessary if you treat `AsyncContext` as a data-only struct (inline all code into file system, and remove the trait `AsyncContext` entirely). i.e.
```
impl SomeFileSystem {
  fn async_read(ctx: &mut Self::Context, handle: Self::Handle, block: FileBlockHandle) {
    handle.submit_async_read(ctx.buf[i], ...)?;
  }
}
```